File: Python/0013.py
class Solution:
    def romanToInt(self, s: str) -> int:
        # Check if the input string is empty
        if not s:
            return 0
        
        # Dictionary to store the integer values of Roman numerals
        roman_to_int = {
            'I': 1, 'V': 5, 'X': 10, 'L': 50,
            'C': 100, 'D': 500, 'M': 1000
        }
        
        # Initialize the total sum
        total = 0
        # Length of the input string
        n = len(s)
        
        # Iterate through the string
        for i in range(n):
            # If current character value is less than next character value, subtract current value from total
            if i < n - 1 and roman_to_int[s[i]] < roman_to_int[s[i + 1]]:
                total -= roman_to_int[s[i]]
            else:
                # Otherwise, add current character value to total
                total += roman_to_int[s[i]]
        
        # Return the total sum
        return total

File: Scala/0013.scala
object Solution {
  def romanToInt(s: String): Int = {
    // Define a mapping from Roman numerals to their integer values
    val romanMap = Map(
      'I' -> 1,
      'V' -> 5,
      'X' -> 10,
      'L' -> 50,
      'C' -> 100,
      'D' -> 500,
      'M' -> 1000
    )

    // Initialize total sum
    var total = 0
    val length = s.length

    // Iterate through the string
    for (i <- 0 until length) {
      // Get the value of the current numeral
      val current = romanMap(s(i))
      // Get the value of the next numeral (if it exists)
      val next = if (i + 1 < length) romanMap(s(i + 1)) else 0

      // Determine if we should add or subtract the current value
      if (current < next) {
        total -= current
      } else {
        total += current
      }
    }

    total
  }
}
